src/join/join.jl

```diff
@@ -273,11 +274,36 @@ function _fill_oncols_left_table_anti!(_res, x, ranges, en, total; threads = tru
     end
 end
 
-function _fill_right_cols_table_left!(_res, x, ranges, en, total, fill_val; threads = true)
-    @_threadsfor threads for i in 1:length(ranges)
-        i == 1 ? lo = 1 : lo = en[i - 1] + 1
-        hi = en[i]
-        length(ranges[i]) == 0 ? _fill_val_join!(_res, lo:hi, fill_val) : copyto!(_res, lo, x, ranges[i].start, length(ranges[i]))
+function _fill_right_cols_table_left!(_res, x, ranges, en, total, fill_val; ra = nothing, inbits = nothing, en2 = nothing, threads = true)
+    if inbits === nothing
+        @_threadsfor threads for i in 1:length(ranges)   
+            i == 1 ? lo = 1 : lo = en[i - 1] + 1
+            hi = en[i]
+            length(ranges[i]) == 0 ? _fill_val_join!(_res, lo:hi, fill_val) : copyto!(_res, lo, x, ranges[i].start, length(ranges[i]))
+        end
+    else
+        @_threadsfor threads for i in 1:length(ranges)
+            i == 1 ? lo2 = 1 : lo2 = en2[i-1] + 1
+            hi2 = en2[i]
+            left_fill_right_col_range!(_res, lo2:hi2, x, ranges[i], en[i], ra[i], inbits)
+        end
+    end
+end
+
+function left_fill_right_col_range!(_res, r2, x, ranges, r, ra, inbits)
+    cnt = 1
+    cnt_r = 1
+    if length(r2) == 0
+        _res = allowmissing(_res)
```

Comment:
this shouldn't be here. If `_res` needs to support `missing`, it must be set when it is initialised.